evm/chains/sharding/mainchain_handler/mainchain_handler.py

```diff
@@ -0,0 +1,117 @@
+import time
+
+import rlp
+
+from web3 import (
+    HTTPProvider,
+    Web3,
+)
+
+from eth_utils import (
+    to_checksum_address,
+)
+
+from evm.chains.sharding.mainchain_handler.config import (
+    DEFAULT_RPC_SERVER_URL,
+    GASPRICE,
+    PASSPHRASE,
+    TX_GAS,
+)
+
+class MainchainHandler:
+
+    def __init__(self, use_eth_tester=False, rpc_server_url=DEFAULT_RPC_SERVER_URL):
+        self._use_eth_tester = use_eth_tester
+        if use_eth_tester:
+            from web3.providers.eth_tester import EthereumTesterProvider
+            from eth_tester import EthereumTester
+            from eth_tester.backends.pyevm import PyEVMBackend
+            self._eth_tester = EthereumTester(
+                backend=PyEVMBackend(),
+                auto_mine_transactions=False,
+            )
+            provider = EthereumTesterProvider(self._eth_tester)
+        else:
+            provider = HTTPProvider(rpc_server_url)
+        self._w3 = Web3(provider)
+        assert self._w3.isConnected()
+
+    # RPC related
+
+    def get_block_by_number(self, block_number):
+        return self._w3.eth.getBlock(block_number)
+
+    def get_block_number(self):
+        return self._w3.eth.blockNumber
+
+    def get_code(self, address):
+        address = to_checksum_address(address)
+        return self._w3.eth.getCode(address)
+
+    def get_nonce(self, address):
+        address = to_checksum_address(address)
+        return self._w3.eth.getTransactionCount(address)
+
+    def import_privkey(self, privkey, passphrase=PASSPHRASE):
```

Comment:
This looks like it's only used for testing.  If so, what do you think about doing this setup directly using the web3 instances during test setup.

Reply:
Thank you! I remove the default value and force the passphrase to be specified when it is called